### ISSUE TITLE
thoth should work inside a job

### ISSUE THREAD

**Issue by jclulow** [joyent/manta-thoth#95]
At the moment, when run in the context of a manta job, thoth exits with this error:

```
jmc@manta # /opt/thoth/bin/thoth upload XXXXX
thoth: expected MANTA_KEY_ID environment variable to be set
```

We should be using the presence of `MANTA_NO_AUTH=true` in the environment to detect that we don't need to `MANTA_KEY_ID` to be set.

**Comment by davepacheco** [comment]
I ran into this again today.  We had a Manta job task dump core.  When that happens, the core file is automatically uploaded to Manta.  It's an out-of-memory dump, so it's pretty large.  To avoid copying it across the WAN just to copy it back into Manta, I was hoping to "thoth upload" it from an "mlogin" session on the job, but that doesn't work:

```
dap@manta # ./node_modules/.bin/thoth upload $MANTA_INPUT_FILE
thoth: expected MANTA_KEY_ID environment variable to be set
```

That's bogus.  MANTA_KEY_ID doesn't need to be set inside of jobs because of the way we implicitly insert user's credentials into outbound requests.

**Comment by chudley** [comment]
As a workaround, @pgale61 found that if you `export MANTA_KEY_ID=foo` inside the mlogin session then thoth will work.

**Comment by bahamat** [comment]
Fixed in e5cd3551220c21afcbd1d9d94773d212e2b33800
